src/mlpack/core/data/load_image_impl.hpp

```diff
@@ -118,30 +116,6 @@ bool Load(const std::vector<std::string>& files,
   return status;
 }
 
-#else // No STB.
-template<typename eT>
-bool Load(const std::string& filename,
-          arma::Mat<eT>& matrix,
-          ImageInfo& info,
-          const bool fatal = false,
-          const bool transpose = true)
-{
-  throw std::runtime_error("Load(): HAS_STB is not defined, "
-      "so STB is not available and images cannot be loaded!");
-}
-
-template<typename eT>
-bool Load(const std::vector<std::string>& files,
-          arma::Mat<eT>& matrix,
-          ImageInfo& info,
-          const bool fatal = false,
-          const bool transpose = true)
-{
-  throw std::runtime_error("Load(): HAS_STB is not defined, "
-      "so STB is not available and images cannot be loaded!");
-}
-#endif // HAS_STB.

```

Comment:
This should be in `load_image.cpp`, since if mlpack itself is compiled without STB support, then these runtime errors should be thrown.